web-js/src/components/useTodos_local.js

```diff
@@ -0,0 +1,67 @@
+import { BSON } from "realm-web";
+import React from "react";
+import { useRealmApp } from "./RealmApp";
+
+const createExampleTodos = (userId = "60810749247a41a9809fba46") => [
+  {
+    _id: new BSON.ObjectID(),
+    _partition: userId,
+    summary: "Do the dishes",
+    isComplete: false,
+  },
+  {
+    _id: new BSON.ObjectID(),
+    _partition: userId,
+    summary: "Buy groceries",
+    isComplete: false,
+  },
+];
+
+function latency(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+export function useTodos() {
+  const realmApp = useRealmApp();
+  const [todos, setTodos] = React.useState([]);
+  
+  const [loading, setLoading] = React.useState(true);
+  React.useEffect(() => {
+    const fetchTodos = async () => {
+      await latency(1640)
+      return createExampleTodos(realmApp.currentUser.id);
+    };
+    fetchTodos().then((t) => {
+      setTodos(t);
+      setLoading(false);
+    });
+  }, [realmApp.currentUser.id]);
+  
+  const saveTodo = async (todo) => {
+    setTodos((t) => [...t, todo]);
+  };
+  const toggleTodo = async (todo) => {
```

Comment:
can you add comments to any methods in this file that are doing slicing, and returning arrays with logic?